### PR TITLE
Fix the race condition in PMIx listener thread.

### PR DESCRIPTION
**Pull request by artpol84**
Thanks to Nysal Jan K A <jnysal@gmail.com> for discoverig this!

### CODE REVIEW

**Comment by mellanox-github** [comment]
* Coverity found 3 errors for pmix
* Coverity report at http://bgate.mellanox.com/jenkins/job/gh-pmix-pr//4/Coverity_Report <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: src/server/pmix_server_listener.c [hunk 1]
```diff
@@ -144,11 +144,13 @@ pmix_status_t pmix_start_listening(struct sockaddr_un *address)
             close(pmix_server_globals.stop_thread[1]);
             return PMIX_ERR_OUT_OF_RESOURCE;
         }
-        /* fork off the listener thread */
+
+	pmix_server_globals.listen_thread_active = true;
+
+	/* fork off the listener thread */
```
Comment (marksantcroos) [inline src/server/pmix_server_listener.c]: Indentation ;-)

**Comment by rhc54** [comment]
@artpol84 We also need the change to the pmix_server_globals struct itself that makes the active flag "volatile"

**Comment by rhc54** [comment]
I went ahead and added the change to the server_globals struct and committed it. Thanks!

**Pull request closed without merging**